src/cache.js

```diff
@@ -1,198 +1,143 @@
-	 * @return {number}
-	 * @constructor
-	 */
-	get WillExpireAt() {
-		return Date.now() + this.aliveDuration;
-	}
-
-	/**
-	 * Get the context for logger().
+	 * getData just retrieve the data from the cache.
 	 *
-	 * @param {Record<string, string>?} customContext The additional context.
-	 * @return {Record<string, string>}
-	 * @private
+	 * It will not update the age of the cache. Only
+	 * for test purpose.
+	 *
+	 * @param {string} key
+	 * @returns {any}
 	 */
-	getLoggerContext(customContext = {}) {
-		return {
-			...customContext,
-			cacheStorageId: this.id,
-		};
+	getData(key) {
+		return this.#lru.get(this.keyOf(key), {
+			updateAgeOnGet: false,
+		});
 	}
 
 	/**
-	 * Remove the expired cache.
+	 * Generate a string key from an object.
+	 *
+	 * @param {any} stuff
+	 * @returns {string}
 	 */
-	removeExpiredCache() {
-		logger.debug(
-			this.getLoggerContext(),
-			'Cleaning up the expired caches...'
-		);
-		this.cacheMap.forEach((cachedData, key) => {
-			if (cachedData.expireAt <= Date.now()) this.cacheMap.delete(key);
-		});
+	keyOf(stuff) {
+		return hash(stuff);
 	}
 
 	/**
 	 * Cache the response.
 	 *
 	 * @template T
-	 * @param {any} key the unique key of action to be cached.
+	 * @param {string?} key the unique key of action to be cached.
 	 * @param {() => Promise<T>} action the action to do and be cached.
-	 * @param {number=} expireAt customize the expireAt of this key.
-	 * @return {Promise<T>}
+	 * @return {Promise<T | null>}
 	 */
-	async cache(key, action, expireAt) {
+	async cache(key, action) {
 		// Disable the cache when the NO_CACHE = true.
 		if (process.env.NO_CACHE === 'true') {
 			return action();
 		}
 
-		// Push the CLEANUP task to the event loop - "polling",
-		// so that it won't block the cache() task.
-		this.emit(CacheStorageEvents.CLEANUP);
-
-		// Check if we have cached it before.
-		// If true, we return the cached value.
-		const cachedData = this.cacheMap.get(key);
-
-		// Object.toString() can't bring any useful information,
-		// we show "Something" instead.
-		const logKey = typeof key === 'object' ? 'Something' : key;
-
-		// Get the logger context with getLoggerContext
-		const logCtx = this.getLoggerContext({
-			logKey,
-		});
-
-		if (cachedData) {
-			logger.debug(logCtx, `${logKey} hit!`);
-			return cachedData.data;
+		// If key is null or undefined, ignore it.
+		if (key == null) {
+			return null;
 		}
 
-		// Cache the response of action() and
-		// register into our cache map.
-		logger.debug(
-			logCtx,
-			`${logKey} did not hit. Storing the execution result...`
-		);
-
-		const sourceResponse = await action();
-		this.cacheMap.set(key, {
-			data: sourceResponse,
-			expireAt: new Date(expireAt || this.WillExpireAt),
-		});
-		return sourceResponse;
-	}
-}
+		const hashedKey = this.keyOf(key);
 
-/**
- * The group which aimed to manage all CacheStorage and
- * call the common method such as `removeExpiredCache()`.
- */
-class CacheStorageGroup {
```

Comment:
`src/app.js` and `src/bridge.js` which still call the old API need to be updated as well.